src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/RuntimeODP.java

```diff
@@ -0,0 +1,422 @@
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("validate successMessage is displayed for the ODP pipeline")
+  public void validateSuccessMessageIsDisplayedForTheODPPipeline() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Get Count of no of records transferred from ODP to BigQuery in {string}")
+  public void getCountOfNoOfRecordsTransferredFromODPToBigQueryIn(String table) throws
+    IOException, InterruptedException {
+    countRecords = gcpClient.countBqQuery(CDAPUtils.getPluginProp(table));
+  }
+
+  @Then("Verify the Delta load transfer is successful in {string} on basis of {string}")
+  public void verifyTheDeltaLoadTransferIsSuccessfull(String table, String field) throws IOException,
+    InterruptedException {
+    int i = 0;
+    String projectId = CDAPUtils.getPluginProp("odpProjectId");
+    String datasetName = CDAPUtils.getPluginProp("dataSetOdp");
+    String selectQuery = "SELECT count(*)  FROM `" + projectId + "." + datasetName + "." + CDAPUtils.getPluginProp
+      (table) + "` WHERE " +
+      field.toUpperCase();
+    switch (datasource) {
+      case "0MATERIAL_ATTR":
+        for (i = 0; i < numList.size(); i++) {
+        Assert.assertTrue(GcpClient.executeQuery(selectQuery.concat("=" + "\"" + numList.get(i) + "\"")) == 1);
+      }
+        countRecords = gcpClient.countBqQuery(CDAPUtils.getPluginProp(table));
+        Assert.assertTrue(countRecords == noOfRecords);
+        break;
+      case "2LIS_02_HDR" :
+        if (action == "I_NUM_C") {
+        for (i = 0; i < numList.size(); i++) {
+          Assert.assertTrue(GcpClient.executeQuery(selectQuery.concat("=" + "\"" + numList.get(i) + "\"")) == 1);
+        }
+        countRecords = gcpClient.countBqQuery(CDAPUtils.getPluginProp(table));
+        Assert.assertTrue(countRecords == noOfRecords);
+        } else {
+        for (i = 0; i < numList.size(); i++) {
+          Assert.assertTrue(GcpClient.executeQuery(selectQuery.concat("=" + "\"" + numList.get(i) + "\"")) == 2);
+        }
+        countRecords = gcpClient.countBqQuery(CDAPUtils.getPluginProp(table));
+        Assert.assertTrue(countRecords == 2 * noOfRecords);
+        }
+        break;
+    }
+
+  }
+
+  @Then("Close the log window")
+  public void closeTheLogWindow() {
+    SeleniumDriver.getDriver().findElement(By.xpath("//*[@data-cy=\"log-viewer-close-btn\"]")).click();
+    numList.clear();
+  }
+
+  @Then("delete table {string} in BQ if not empty")
+  public void deleteTableInBQIfNotEmpty(String table) throws IOException, InterruptedException {
+    try {
+      int existingRecords = gcpClient.countBqQuery(CDAPUtils.getPluginProp(table));
+      if (existingRecords > 0) {
+        gcpClient.dropBqQuery(CDAPUtils.getPluginProp(table));
+        BeforeActions.scenario.write("Table Deleted Successfully");
+      }
+    } catch (Exception e) {
+      BeforeActions.scenario.write(e.toString());
```

Comment:
should we fail the test if an exception was thrown?

Reply:
No we should not fail as if the table is not found in the catch block it will return table not found and we have written the same in the report.